pkg/controllers/deleting/controller.go

```diff
@@ -110,14 +121,34 @@ func NewController(
 	return c
 }
 
+func newDeletignMetrics(logger logr.Logger) deletingMetrics {
```

Comment:
this change looks wrong to me, it basically reverts the PR which changed this the other way around. 

Metrics need to be handled in the metrics package only